### ISSUE TITLE
3.1.3: pytest is failing

### ISSUE THREAD

**Issue by kloczek** [python-lz4/python-lz4#222]
I'm trying to package your module as rpm packag. So I'm using typical in such case build, install and test cycle used on building package from non-root account:
- "setup.py build"
- "setup.py install --root </install/prefix>"
- "pytest with PYTHONPATH pointing to sitearch and sitelib inside </install/prefix>

May I ask for help because few units are failing in one unit
```console
+ PYTHONPATH=/home/tkloczko/rpmbuild/BUILDROOT/python-lz4-3.1.3-2.fc35.x86_64/usr/lib64/python3.8/site-packages:/home/tkloczko/rpmbuild/BUILDROOT/python-lz4-3.1.3-2.fc35.x86_64/usr/lib/python3.8/site-packages
+ /usr/bin/pytest -ra
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.8.11, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
Using --randomly-seed=194221891
rootdir: /home/tkloczko/rpmbuild/BUILD/python-lz4-3.1.3, configfile: tox.ini
plugins: forked-1.3.0, shutil-1.7.0, virtualenv-1.7.0, expect-1.1.0, flake8-1.0.7, timeout-1.4.2, betamax-0.8.1, freezegun-0.4.2, aspectlib-1.5.2, toolbox-0.5, rerunfailures-9.1.1, requests-mock-1.9.3, cov-2.12.1, pyfakefs-4.5.0, flaky-3.7.0, benchmark-3.4.1, xdist-2.3.0, pylama-7.7.1, datadir-1.3.1, regressions-2.2.0, cases-3.6.3, xprocess-0.18.1, black-0.3.12, anyio-3.3.0, asyncio-0.15.1, trio-0.7.0, httpbin-1.0.0, subtests-0.5.0, isort-2.0.0, hypothesis-6.14.6, mock-3.6.1, profiling-1.7.0, randomly-3.8.0, Faker-8.12.1
collected 31996 items

tests/stream/test_stream_4.py ..                                                                                                                                     [  0%]
tests/frame/test_frame_0.py .................                                                                                                                        [  0%]
tests/stream/test_stream_0.py ss..s.ss.s............s...s.s...s..s...s...ss..s...ss.s..sss..s........s.s.......s.s...ss.......sss......................s..s.......s. [  0%]
....ss..s........s..s.........s..s.....s..s....s..s.......s...sss.........ss......s.ss.s......ss......s........s....s.sss.sss....s.................s..s..s..s..ss..s [  0%]
....s...s..s.....s..........s...........s....s...s.ss.s.s.s.....ss..s..s.....s...........s....s......s........s......s....s.ss........sss..s.....s..............s..s [  1%]
..s...s........s.s.s.s.....s...........ss....ss..s.....ss...s..s........s..s.ss..s..s....s....s.ss..s..s......s.....ss.s.........s.....ss...s.s...........ss....ss.. [  2%]
.ssss..ss.....s..........ss..ss......................s...............s..s...s.............s........s....s.....s..s.......sss.s....s.s...s..........s.......s..s..... [  2%]
......ss.....s..s...s............s............sss..s.s...s..sss.....s...s.....sss.ssss.s....s..s.......s..s.........s.ss...s...s.s...s....s......s.s...............s [  3%]
............s....s.s.....s.s..........ss.......s..s........ss.........s....s......s.s...............s...ss.s.s..s.............ssss..s.......................ss.s.... [  3%]
.s.s................s.....s.s.s.....ss...ss...s..s....s...s.............s..s....s.s..........s.......s........s......s..s.........s.....s.........s.s......s..ss.s.s [  4%]
..sss..........ss.s.s..s...s.s...ss.....sss.....ss...............s..s.ssss....ss.....s..............s.ss.....s....ss.s..s..s.....s.....s..ss.ss.s..s.........s....s. [  4%]
..s..s...ss...s.........s..s........sss.....s...s..s......s....s........s...s.......ss.s...s.s........s...s..sssss..s...ss..ss.s...s.....s.............sss...s...... [  5%]
.s......s.s..ss......ss.....s........s...s.s..s.ss.....s..s....s..s.....s..s..s.........s..s..s.....sss.s..s....s.............s........s..s.s.ss..s......s.......s.. [  5%]
.s.s.....s.....s.s........s..s.s....s....s.....s..s...s........s...s............s..s.....s.s..s.s...s..ss.ss..........sss.s......s...ss..s.s..s.s.................ss [  6%]
.s..s..s.....s.....ss..........s............ss.....s..s.s.s....s.sss.s...........s.....s........ss......s.s.....s........s..s....s..ss.s............ssss...s......s. [  6%]
.......s....s....s.....s....ssss.......s...s................ss.....s.s.......s....s..s......s..s..s.s..........s..ss.........s...s.s...............s........s.s..... [  7%]
............ss..ss.....s.ss..s......s..s...s......s......ss....s..s....s...s.s.s...s..s......ss....s...s.........s..........ss..s...............s.s.s....s..s.s..... [  7%]
...ss............s............sss..s...s...s.s.s..ss..s..s...........s.....s.......s.s.ss..s..s.s...s.......s.s.sss..s.......s..s.........s.ss..s...............s..s [  8%]
s..s.....s......s..s...s......s.s.s..s.ss..s.s.............ss........s....ss.s..s.s.s....s.s..ss.s..........s...s.......s....ss................s..s...s.........s..s [  8%]
.s......ss....s..............ss........ss..s...s........sss..ss.ss.........s.........s....s......ssss....s....s..s.........s....s......ss........s.s.s...s.....s...s [  9%]
............................s....s........s.........ssssss...s..s.....s.s.s...s.......ss.....s..s......ss......s.s.....s...sssss...s..s..................s...ss..... [  9%]
.ss.s..s.........s......ss.....s............s.....s...s.s..s..s.s...s..s.s..s..s.s.s.s..s..s..........s......s............................s.s.ss.s......s....s..ss.s [ 10%]
s.s.ss...s....s...s...s..s.ss.s............................s..ss.....s..s.....s.........s..........s......sss..s.s...s..s.....s.ss.....s.......ss......s..s..s....s. [ 10%]
..sss.s......s.....s........s..........s.........s...ss......ss...s..s......s.s.s......s..ss.............s.ss.....ss.s.....s..........s...s......s...........s...... [ 11%]
.ss.s...ss..s.......s..s..s..........s..s.....s....s...s..s...ss.........ss.s.s...........ss.s......s.......s............s........ss.....s............s..s.ss.s.s... [ 11%]
s.....s.....s................s..ss.s.s..s..........s..s...s..ss...s..ss..s...s...........s..s.s......s.....s...............ss.......s.s...s..s......s.s....s........ [ 12%]
.s.sss.s......ss.ss...............s...ss.........s..............s...........s...s...s.....ss.....s.......s..ss...s..s.ss.s.s...s..ss...s...ss.s......s..s........... [ 12%]
.sss............s...sss.....s.s.s.s.....s....s.s........ss.....s.s....s..s..s.ss.s.........ss.s...s............sss..s......s.s.....s.s.sss.....s.s.......s...sss..s. [ 13%]
..s...........ss....s.........s..s...s.s.........s.....s.....s......s....s.ss..s..s...sss.ss.......s....s............s...s.................ss.ss..ss.....s....s.ss.. [ 13%]
s...........s.....s.......sss..ss....s.....s.s.s.......sss...ss...ss......s..sss.s.......ss..ss...s........s..s.ss...s....s.......s..ss............s.s.ss....s.....s [ 14%]
s..s...........s.s..ss.......s..s.......s..s....s.s.....s...........s........s......s......s........s..s....s......ss.s.......s.........ss....s..s.........ss..s.... [ 14%]
.s.ss....ss.....s...ss...s.....................s.s...s..ss..s...s...sss.s.s..s...s...........s....ss.s......s...s.....ss.....s...ss...s................s..s.s.....s. [ 15%]
s...s...........s..s....s..s...s...s..sss.......s.ss...s.........s...s......s..s.......ss.s.......s.....s.s........s...sssss.....ss.ss..ssss..s.s...s.......s...s... [ 15%]
.....ss.........s....s........s........s....ss.s.....s...s.s..............s...s.....s.....s..ss.s.s.......s..ss...s.....s.....s...........s......s.s.....s.......... [ 16%]
.s......sss.s....s..........s.....s...s.s....ss...s..s.....s..........s......s.........s......s.ss.s............s..s...s....s..........s..ss.........s........s....s [ 16%]
........s.s.ss.s.................s..s..sss...s....s.s....s.............s..s..ss..ss.ss..ss.........s......ss.s..s.........s.s....s.............ss.......ss.........s [ 17%]
..s.s..ss.s.......s.......s........s....s.............sss.s........s......s........s.s.s...s...s...s....s..s..s.s....s.....s.....s.s...ss..................s..ss.... [ 17%]
s......s.....s...........sssss...s...s..........s.s...s.....s...s.s.ss....s..s....sss..s..........ssss..s.s....s...s...sss...ss............s..s.....s....ss.sss..... [ 18%]
.s.s.s....ss.........s...........s..sss..s.s...ssss.ss..sss.......s......s..s.....s..ss.ss....sss..s.s....s..s.....s.......s.s.s...s....s......s..s........ss......s [ 18%]
......sss.....ss..s.................sss..s.s.....ss.ss............ss...........ss..s..s...s........s....................s.s........s.s.............s...s......ssss.. [ 19%]
....s..ss...s..s.........s...............s...s.s....s..s......s.s................s..s....ss.....s...ss.sss..s...ss....ss.s....s.....s..s.......s....sss.s..ss.....s. [ 19%]
..s.ss..s.....s..s.........ss......s...s....s...s....s......ss.....s....ss................s................s..s.s..s......s..s..s...s........s...s..sss.s.s.s....... [ 20%]
...ss..s.............ss..s......ss.......s...................s...ss......s.......s......s......s.s.s.s...s.....s.s...............s.s....s...s...s........s..s...s... [ 20%]
...........s......s.....s..s..s.ss......s...sss..ss.......s...sss..........s.............s...ss..s.......s......s..s..........s...........s...s....s...s....s....... [ 21%]
.s.......s.....s....s..s.....ss.......s.ss......ss..s.s..........s.ss.ss...s..s.....s........s...s.....ss....s..ss.s...s...ss.....s......s..ss..............ss....s. [ 22%]
......s........s......s.ss.....s...ss...s...s.ss......s...ssss..s..s..............s......ss.............s..sss.........s.....ss.ss....ss..............s......s.s.... [ 22%]
s...s..s...s......s.ss.....ss..s....s.....s.s......ss..s......ss.s....s.s.s............s.s...s....s.......sss............s...............s.s..s................s.... [ 23%]
..sss....s............ss.......s...s...sss............s...........s.....s....ss....ss.s..s.........s..s.s...............s.......s.s...s...s.......s.s...s..s......ss [ 23%]
......s...sss..ss.s......s.ss.s...s..s.s.s..s.s.ss.........s...sss..s.ss.s..s..s......ss..s..s.s.....s......s......s..ss.....s......s........s.....s........s.ss.s.. [ 24%]
.s..s......ss........s.....s...s..ss......ss..s.s....s..s..s...ss.....s.s.....s...........s....s.ss.ss...............s...s.......s..ss.s.ss.....s....s....s.......s. [ 24%]
.................s..s......sss.s......s...s.s...s.......s.....s.....s...........s..ss......s....s.....s...........s..ss.............s..s.ss.s..ssss............s.... [ 25%]
s....sss.......s.s..s....s...s.....ss...s..s.......s.s.s........s.s............s.....s.......s...sss..s........s...ss..ss..s....ss.ss.............s.....s....s...s.. [ 25%]
.......s.....s..s..s...s...s.s.....s......ssss.........s.............s....s.s.ssss.......s.ssss.s...................s...s......ss.s..s..........sss..s.s.....s...... [ 26%]
....s......ss....ss.........s...s................s....s.......sss.s.......s..ss..s....s..sss...........s...s.s.....ss.....ss.s.s....s.s........s......s........sss.s [ 26%]
....s...ss........s.s.s.......ssss.s..s...s.s..s....ss.s.sss.s....s..s.....s...sss.sss..s..s.........s.s..s..s.s....ss...ss.ss.......s.s.s....s.ss.....ss.s....ss... [ 27%]
s...........s.ss....s.....ss...........s..........s.......s.....s.....s.s..s....s....s..s......s....s....s...........s.....s.....s..........s.....s.......s.s..s.s.s [ 27%]
s...s..s.................s....s..s..s..s.......ss.s.....ss....ss....s.s......s.s................s.........s..ss..........s..s.......s..s.s....s.....s...s...s....s.s [ 28%]
..s..s...........s.......ss...............s..ss...s.s....s...............s.....s...ss..s...s..........s........ss...................s.......s.....s.ss.ss.s...s..... [ 28%]
s..s...s.........sss.ss.s...s.......s.ss...s.........s.........ss....s....s.....s..............s.s........s.s.ss.ss..............s.........s...............s.s....s. [ 29%]
.......ss.....s........s....s..ss.....ss.s...s.s....s..s.......s..s.ss......s...s.ss..........s........s.s.s.............s..s...s....ss...s.s..ss................... [ 29%]
.s....s...s......s......s......s...ss....ssss.s..s...s......s.....s..sss..ss.....ss...s........s.s..s...........sssss......s...s....s..s........s..s....s.s...s.s... [ 30%]
.s..s.....s.....ss...........s...s.......ss....s...s..s......s....s..............s....s...s..s....s.......ss.ss.s.........s.....s......s......s......ss.ss.ss....... [ 30%]
.............................s...............s...ss...s.ss.s..s..s.............s.s....ss..............s..ss......s.s.s..sss.............s.....s...s.ss...........s.. [ 31%]
...s.....s...........................ss......s........s......s...s....s....s.s.......s...s......s..s.......s............s..s..sss.............ss.............s...... [ 31%]
s..s...s....ss...ssss.....s..ss......ss...ss.........s.......ss.s.ss..ss.....s..ss..s..............s....s.s.......s..s.....s..s..s.....s..s........s.s....s.....s... [ 32%]
....s..s....s.......ss.....s.s..s....s..s..s.s.sss.....sss..s..........s....s..s.ss...s...s...ss......s.ss.....s.s...s...s...s.....s.......s.s.s............s..s.... [ 32%]
.....ss..s.......s..s..s.....s...s.s....s........ss....s....s.......s.......s.........s.ss.....s...s.s.ss....s.............s..s..s......ss..........s.s......s.....s [ 33%]
....s....s...s.s.............s..s..s.s.s...s...s.s..........s...s..s.s....s.ss.....s.ss.......s..s......ss..s.s.s................s..s.....ss.s...............s..s.s. [ 33%]
s....s.....ss..s.......s.ss.s....s......s.s.....s......s.....s................s...............s.s....s....s.............s...s......s.ss.....s...ss........s......... [ 34%]
s......s.s.......s........s........s.s.s.........s.s.ss...ss....s...s...s............s..s.........s.....s......ss..........s.s...ss.......s....s.s..............s... [ 34%]
..............ss.s...s.....s.....ss.ss..s.s...s.sss...............s....ss......s...s.ss....s.s.s.s..s....ss...s....sss.................s...s.............s.......s.s [ 35%]
...s.s.s......s...s..................ss.............s....ss.....ss..ss..s.....s..s...........s........s..s.......s..............s.ss.............s............s..... [ 35%]
....s......s.......s.....s..s.s.s......................s..s.s......s.........s.......s......ssss..s...ss......s...s..ss......ss..s.......ss......s.s....s........... [ 36%]
....s...ss.........s......s......s.s..............s..s...s.s..ss.......s.s.....s.....s........s...............s..s...s.........s.....ss.....s....s..ss.s.s..s.....s. [ 36%]
.s...s.s..............ss...s.......s.....s....ss....s......s..........s..s.............ss.s.....s.....sss.s......ss.ss......s....s..s.ss......s.ss.........s.....ss. [ 37%]
....ss.....ss.s.......sss...ss.s...s....s......s.s.s......s.s....s.......s............s.s..s.s.....ss....s.....sss.......s..s.s.s.....s..ss..........s..s..ss.s.s... [ 37%]
.....ss..ss.s....s........ss..s.s..s..ss....                                                                                                                         [ 38%]
tests/stream/test_stream_3.py ......                                                                                                                                 [ 38%]
tests/frame/test_frame_8.py .                                                                                                                                        [ 38%]
tests/frame/test_frame_2.py ........................................................................................................................................ [ 38%]
.................................................................................................................................................................... [ 38%]
.................................................................................................................................................................... [ 39%]
.................................................................................................................................................................... [ 40%]
.................................................................................................................................................................... [ 40%]
.................................................................................................................................................................... [ 41%]
.................................................................................................................................................................... [ 41%]
.................................................................................................................................................................... [ 42%]
.................................................................................................................................................................... [ 42%]
.................................................................................................................................................................... [ 43%]
.................................................................................................................................................................... [ 43%]
.................................................................................................................................................................... [ 44%]
.................................................................................................................................................................... [ 44%]
.................................................................................................................................................................... [ 45%]
.................................................................................................................................................................... [ 45%]
.................................................................................................................................................................... [ 46%]
.................................................................................................................................................................... [ 46%]
.................................................................................................................................................................... [ 47%]
.................................................................................................................................................................... [ 47%]
.................................................................................................................................................................... [ 48%]
.................................................................................................................................................................... [ 48%]
........................................                                                                                                                             [ 48%]
tests/frame/test_frame_4.py ........................................................................................................................................ [ 49%]
.................................................................................................................................................................... [ 49%]
.................................................................................................................................................................... [ 50%]
.................................................................................................................................................................... [ 50%]
.................................................................................................................................................................... [ 51%]
.................................................................................................................................................................... [ 51%]
.................................................................................................................................................................... [ 52%]
.................................................................................................................................................................... [ 52%]
.................................................................................................................................................................... [ 53%]
.................................................................................................................................................................... [ 53%]
.................................................................................................................................................................... [ 54%]
.................................................................................................................................................................... [ 54%]
.................................................................................................................................................................... [ 55%]
.................................................................................................................................................................... [ 55%]
.................................................................................................................................................................... [ 56%]
................................................................................................................................                                     [ 56%]
tests/frame/test_frame_6.py ........................................................................................................................................ [ 57%]
.................................................................................................................................................................... [ 57%]
.................................................................................................................................................................... [ 58%]
.................................................................................................................................................................... [ 58%]
....................................................................................................................................................                 [ 59%]
tests/frame/test_frame_3.py ...                                                                                                                                      [ 59%]
tests/stream/test_stream_1.py .................................                                                                                                      [ 59%]
tests/frame/test_frame_7.py ....                                                                                                                                     [ 59%]
tests/frame/test_frame_5.py ....                                                                                                                                     [ 59%]
tests/block/test_block_2.py ..                                                                                                                                       [ 59%]
tests/block/test_block_3.py F

================================================================================= FAILURES =================================================================================
__________________________________________________________________ test_block_decompress_mem_usage[data0] __________________________________________________________________

data = b'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'

    def test_block_decompress_mem_usage(data):
        tracemalloc = pytest.importorskip('tracemalloc')

        tracemalloc.start()

        compressed = lz4.block.compress(data)
        prev_snapshot = None

        for i in range(1000):
            decompressed = lz4.block.decompress(compressed)  # noqa: F841

            if i % 100 == 0:
                snapshot = tracemalloc.take_snapshot()

                if prev_snapshot:
                    stats = snapshot.compare_to(prev_snapshot, 'lineno')
>                   assert stats[0].size_diff < (1024 * 4)
E                   AssertionError: assert 4544 < (1024 * 4)
E                    +  where 4544 = <StatisticDiff traceback=<Traceback (<Frame filename='/usr/lib64/python3.8/tracemalloc.py' lineno=532>,)> size=4544 (+4544) count=94 (+94)>.size_diff

compressed = b'\x00\x00\x10\x00\x1fa\x01\x00\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xf...\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xf6Paaaaa'
data       = b'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
decompressed = b'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
i          = 100
prev_snapshot = <tracemalloc.Snapshot object at 0x7f9b29e0d760>
snapshot   = <tracemalloc.Snapshot object at 0x7f9b29e0d880>
stats      = [<StatisticDiff traceback=<Traceback (<Frame filename='/usr/lib64/python3.8/tracemalloc.py' lineno=532>,)> size=4544 (...aceback=<Traceback (<Frame filename='/usr/lib64/python3.8/warnings.py' lineno=445>,)> size=80 (-64) count=1 (-1)>, ...]
tracemalloc = <module 'tracemalloc' from '/usr/lib64/python3.8/tracemalloc.py'>

tests/block/test_block_3.py:36: AssertionError
========================================================================= short test summary info ==========================================================================
SKIPPED [2700] tests/stream/test_stream_0.py:102: Invalid case: buffer_size too large for the block length area
FAILED tests/block/test_block_3.py::test_block_decompress_mem_usage[data0] - AssertionError: assert 4544 < (1024 * 4)
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================== 1 failed, 16314 passed, 2700 skipped in 198.81s (0:03:18) =========================================================
pytest-xprocess reminder::Be sure to terminate the started process by running 'pytest --xkill' if you have not explicitly done so in your fixture with 'xprocess.getinfo(<process_name>).terminate()'.
```

**Comment by jonathanunderwood** [comment]
This test is a very crude check to see if there are any memory leaks from the C modules. You can reasonably increase the `(1024 * 4)` to `(1024 * 8)` here, and the tests will pass. PR welcome.

**Comment by jonathanunderwood** [comment]
This should be fixed in 4.0.0
